### PR TITLE
Fix KMyMoney macOS builds 

### PR DESCRIPTION
**Pull request by wrobelda**
Fixes a bunch of issues that caused macOS builds to fail. Tested locally with a freshly checked-out craft and the .dmg is building fine. 

It should also fix broken AqBanking for Windows builds as well

### CODE REVIEW

File: libs/_autotools/aqbanking/aqbanking.py [hunk 1]
```diff
@@ -28,11 +28,11 @@
 
 class subinfo(info.infoclass):
     def setTargets(self):
-        self.targets["5.7.8"] = "https://www.aquamaniac.de/rdm/attachments/download/27/aqbanking-5.7.8.tar.gz"
-        self.targetDigests["5.7.8"] = (['16f86e4cc49a9eaaa8dfe3206607e627873208bce45a70030c3caea9b5afc768'], CraftHash.HashAlgorithm.SHA256)
-        self.targetInstSrc["5.7.8"] = "aqbanking-5.7.8"
-        self.defaultTarget = "5.7.8"
-        self.patchLevel["5.7.8"] = 1
+        self.targets["6.1.2"] = "https://www.aquamaniac.de/rdm/attachments/download/269/aqbanking-6.1.2.tar.gz"
+        self.targetDigests["6.1.2"] = (['f94c9197302fe73344261a779f8fa4d9'], CraftHash.HashAlgorithm.MD5)
```
Comment (TheOneRing) [inline libs/_autotools/aqbanking/aqbanking.py]: please no md5

File: libs/_autotools/gwenhywfar/gwenhywfar.py [hunk 2]
```diff
@@ -28,15 +28,15 @@
 
 class subinfo(info.infoclass):
     def setTargets(self):
-        self.targets["4.20.0"] = "https://www.aquamaniac.de/rdm/attachments/download/10/gwenhywfar-4.20.0.tar.gz"
-        self.targetDigests["4.20.0"] = (['5a88daabba1388f9528590aab5de527a12dd44a7da4572ce48469a29911b0fb0'], CraftHash.HashAlgorithm.SHA256)
-        self.targetInstSrc["4.20.0"] = "gwenhywfar-4.20.0"
+        self.targets["5.2.0"] = "https://www.aquamaniac.de/rdm/attachments/download/256/gwenhywfar-5.2.0.tar.gz"
+        self.targetDigests["5.2.0"] = (['cda539d33571000c0db368266b34e74e'], CraftHash.HashAlgorithm.MD5)
```
Comment (TheOneRing) [inline libs/_autotools/gwenhywfar/gwenhywfar.py]: please no md5

File: libs/_autotools/aqbanking/aqbanking.py [hunk 3]
```diff
@@ -28,11 +28,11 @@
...
+        self.targetInstSrc["6.1.2"] = "aqbanking-6.1.2"
+        self.defaultTarget = "6.1.2"
+        self.patchLevel["6.1.2"] = 1
```
Comment (TheOneRing) [inline libs/_autotools/aqbanking/aqbanking.py]: no need for a patch lvl after an update

File: libs/_autotools/aqbanking/aqbanking.py [hunk 4]
```diff
@@ -64,14 +64,10 @@ def postInstall(self):
                         if craftRoot.endswith("/"):
                             craftRoot = craftRoot[:-1]
                         cmakeFileContents[i] = cmakeFileContents[i].replace(m.group('root'), craftRoot)
-
-                    m2 = re.search("libaqbanking.so", cmakeFileContents[i])
-                    if m2 is not None:
-                        cmakeFileContents[i] = cmakeFileContents[i].replace("lib/libaqbanking.so", "bin/libaqbanking-35.dll")
                 elif CraftCore.compiler.isMacOS:
```
Comment (TheOneRing) [inline libs/_autotools/aqbanking/aqbanking.py:66]: we had an issue with symlinks before, is that step needed? a symlink libaqbanking.so  -> libaqbanking.so.ver should exist nowadays.

**Review by TheOneRing — CHANGES_REQUESTED** [review]

File: libs/_autotools/aqbanking/aqbanking.py (hunk 4 above)
Comment (wrobelda) [reply libs/_autotools/aqbanking/aqbanking.py:66]: The earlier aqbanking 5.3.8 only had a single line in its cmake which assumed `.so` extension irregardless of compilation platform: `set_and_check(AQBANKING_LIBRARIES "@libdir@/libaqbanking.so")`. This is why the blueprint contained a patching logic for both win32 and macOS to lookup their respective extensions (`dll` and `dylib`). 

The 6.1.2, however, at least handles the `.dll` extension properly now: 
```
if(WIN32)
    set_and_check(AQBANKING_LIBRARIES "@libdir@/libaqbanking.dll.a")
else()
    set_and_check(AQBANKING_LIBRARIES "@libdir@/libaqbanking.so.@AQBANKING_SO_EFFECTIVE@")
endif()
```
I assumed this works as expected, so I removed the patching logic for win32 altogether. The one for `dylib`, however, is still required.

File: libs/_autotools/aqbanking/aqbanking.py (hunk 1 above)
Comment (wrobelda) [reply libs/_autotools/aqbanking/aqbanking.py]: Fixed.

File: libs/_autotools/gwenhywfar/gwenhywfar.py [hunk 5]
```diff
@@ -28,15 +28,15 @@
...
         if CraftCore.compiler.isMinGW():
-            self.patchToApply["4.20.0"] = [("gwenhywfar-4.19.0-20180218.diff", 1)]
+            self.patchToApply["5.2.0"] = [("gwenhywfar-4.19.0-20180218.diff", 1)]
         elif CraftCore.compiler.isMacOS:
-            self.patchToApply["4.20.0"] = [("gwenhywfar-4.20.0-20180503.diff", 1)]
-        self.defaultTarget = "4.20.0"
-        self.patchLevel["4.20.0"] = 3
+            self.patchToApply["5.2.0"] = [("gwenhywfar-4.20.0-20180503.diff", 1)]
+        self.defaultTarget = "5.2.0"
+        self.patchLevel["5.2.0"] = 3
```
Comment (TheOneRing) [inline libs/_autotools/gwenhywfar/gwenhywfar.py:39]: patch lvl can be removed with the update.

**Review by TheOneRing — APPROVED** [review]
One nitpick.
Do you have access to the KDE git?
We can't use github to merge changes.
If you don't have access I can do it for you.

**Comment by wrobelda** [comment]
> One nitpick.
> Do you have access to the KDE git?
> We can't use github to merge changes.
> If you don't have access I can do it for you.

I don't and would appreciate it.

**Pull request merged**